src/Entities/Entity.cpp

```diff
@@ -222,8 +222,15 @@ void cEntity::Destroy(bool a_ShouldBroadcast)
 	}
 
 	cChunk * ParentChunk = GetParentChunk();
-	m_World->QueueTask([this, ParentChunk](cWorld & a_World)
+
+	// We remove the entity from memory 5 ticks later, giving everyone plenty of time
+	// to read IsTicking(formerly IsDestroyed), finding out it's false, and nulling out their pointers.
+	// LogicParrot TODO Clear out the previous much more complicated mechanism
+	// (See: cPawn::StopEveryoneFromTargetingMe() and its friends)
+	// Take world travel into account.
+	m_World->ScheduleTask(40, [this, ParentChunk](cWorld & a_World)
```

Comment:
Yes. magic, but 2 should be enough. Here's the deal (Also see my response to @tigerw )

`cMonster::m_Target` is an entity pointer, and it used to cause trouble when mobs are freed from memory or teleported to new worlds because other entities still pointed at them. The solution was some relatively complex set of functions that nullified m_Target when needed, (cPawn::stopEveryoneFromTargetingMe, etc.). In this PR, there are new entity pointers like `m_Attacker`  because the code is split into modular behaviors so the old technique doesn't cover them - and seems needlesly complex anyways -. The new technique is much simpler: Do not free entities from memory immediately when they are destroyed, just wait a bit. During the wait, everyone holding a pointer to that entity would call `If (!myPointer.IsTicking()) myPointer = nullptr`. 2 ticks should be enough, assuming everyone makes this check once per tick. (1 tick isn't enough because if A is freed prior to B and B makes the check, then B just used a dangling pointer).